src/main/java/task/Deadline.java

```diff
@@ -0,0 +1,148 @@
+        formatters.add(DateTimeFormatter.ofPattern("dd/MMM/yyyy HH:mm:ss"));
+        formatters.add(DateTimeFormatter.ofPattern("dd/MMM/yyyy HH:mm:ss.SSS"));
+        formatters.add(DateTimeFormatter.ofPattern("dd-MMM-yyyy HH:mm"));
+        formatters.add(DateTimeFormatter.ofPattern("dd-MMM-yyyy HH:mm:ss"));
+        formatters.add(DateTimeFormatter.ofPattern("dd-MMM-yyyy HH:mm:ss.SSS"));
+        formatters.add(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm"));
+        formatters.add(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss"));
+        formatters.add(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss.SSS"));
+        formatters.add(DateTimeFormatter.ofPattern("yyyy-MMM-dd HH:mm"));
+        formatters.add(DateTimeFormatter.ofPattern("yyyy-MMM-dd HH:mm:ss"));
+        formatters.add(DateTimeFormatter.ofPattern("yyyy-MMM-dd HH:mm:ss.SSS"));
+        formatters.add(DateTimeFormatter.ofPattern("yyyy/MM/dd HH:mm"));
+        formatters.add(DateTimeFormatter.ofPattern("yyyy/MM/dd HH:mm:ss"));
+        formatters.add(DateTimeFormatter.ofPattern("yyyy/MM/dd HH:mm:ss.SSS"));
+        formatters.add(DateTimeFormatter.ofPattern("yyyy/MMM/dd HH:mm"));
+        formatters.add(DateTimeFormatter.ofPattern("yyyy/MMM/dd HH:mm:ss"));
+        formatters.add(DateTimeFormatter.ofPattern("yyyy/MMM/dd HH:mm:ss.SSS"));
+
+        deadlineString = deadlineString.trim();
+        if (!deadlineString.contains(":")) { // No time
+            deadlineString = deadlineString + " 00:00";
+        }
+
+        for (DateTimeFormatter formatter : formatters) {
+            try {
+                LocalDateTime deadlineDateTime = LocalDateTime.parse(deadlineString, formatter);
+                return deadlineDateTime;
+            } catch (DateTimeParseException e) { // task.Deadline is not a DateTime
+                // do nothing
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        sb.append("[D]");
+        sb.append("[" + getStatusIcon() + "] ");
+        sb.append(description);
+        sb.append(" (by: " + displayTime() + ")");
+        return sb.toString();
+    }
+
+    /**
+     * Checks if it is a deadline, to differentiate between tasks
+     *
+     * @return true
+     */
+    @Override
+    public boolean isDeadline() {
+        return true;
+    }
+
+    /**
+     * @return A boolean on whether deadline is overdue or not
+     */
+    public boolean overdue() {
+        if (isDateTime) {
+            if (deadlineDateTime.isAfter(LocalDateTime.now())) {
+                return false;
+            } else {
+                return true;
+            }
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * Returns a boolean of whether the deadline is due by the specified date
+     * @param localDateTime Date and Time of the deadline
+     * @return boolean
+     */
+    public boolean dueBy(LocalDateTime localDateTime) {
```

Comment:
Similar issue to overdue, this time it could be something like `isDueBy`? That way it would feel more like a question, ie is `[task name]` due by `[localDateTime param]`?